test/test_unary_ufuncs.py

```diff
@@ -19,9 +158,39 @@
 # See NumPy's universal function documentation
 # (https://numpy.org/doc/1.18/reference/ufuncs.html) for more details
 # about the concept of ufuncs.
+
+# TODO: port test_unary_out_op_mem_overlap
+# TODO: add out= tests (different devices, dtypes, mismatched sizes,
+#                       correct sizes, 0 size, broadcasted out)
+# TODO: add test for inplace variants erroring on broadcasted inputs
 class TestUnaryUfuncs(TestCase):
     exact_dtype = True
 
+    # Helper for comparing torch tensors and numpy arrays
+    # TODO: should this or assertEqual also validate that strides are equal?
```

Comment:
Given how complicated our layout propagation rules are, it seems unlikely that Numpy is going to exactly follow  them 👍

Reply:
Depends on how adversarial we want to be with the inputs here, for many cases there is an obvious and correct answer.

Reply:
Speaking of this, here's a case where I propose to deviate from numpy behavior:
```
In [1]: import numpy as np                                                                                                                                                                                         
In [2]: a=np.arange(24).reshape(6,4,1,1).transpose(0,3,1,2)                                                                                                                                                        
In [3]: a.shape                                                                                                                                                                                                    
Out[3]: (6, 1, 4, 1)
In [4]: a.strides                                                                                                                                                                                                  
Out[4]: (32, 8, 8, 8)
In [5]: a.dtype                                                                                                                                                                                                    
Out[5]: dtype('int64')
In [6]: b=np.array([1], dtype=np.int64)                                                                                                                                                                            
In [7]: (a+b).strides                                                                                                                                                                                              
Out[7]: (32, 32, 8, 8)
```
I think that in this case output strides should be equal to full-size input (`a`) strides.
That said, all divergent behavior should be limited to strides at the dimensions of size 1.